flink-core/src/main/java/org/apache/flink/configuration/BatchExecutionOptions.java

```diff
@@ -99,7 +99,7 @@ public class BatchExecutionOptions {
     public static final ConfigOption<Integer> ADAPTIVE_AUTO_PARALLELISM_DEFAULT_SOURCE_PARALLELISM =
             key("execution.batch.adaptive.auto-parallelism.default-source-parallelism")
                     .intType()
-                    .defaultValue(1)
+                    .defaultValue(0)
                     .withDeprecatedKeys(
```

Comment:
Why do you need to change the default value? I think a better way is through  `configuration.getOptional(BatchExecutionOptions.ADAPTIVE_AUTO_PARALLELISM_DEFAULT_SOURCE_PARALLELISM))`
to determine whether the configuration item has been configured